packages/grafana-ui/src/components/QueryField/QueryField.story.tsx

```diff
@@ -1,18 +1,57 @@
 import { ComponentMeta, ComponentStory } from '@storybook/react';
 import React from 'react';
 
-import { QueryField } from '@grafana/ui';
-
+import { TypeaheadInput } from '../../types';
 import { withCenteredStory } from '../../utils/storybook/withCenteredStory';
 
+import { QueryField, QueryFieldProps } from './QueryField';
+
+const disabledControl = {
+  table: {
+    disable: true,
+  },
+};
+
 const meta: ComponentMeta<typeof QueryField> = {
   title: 'Data Source/QueryField',
   component: QueryField,
   decorators: [withCenteredStory],
+  argTypes: {
+    onTypeahead: disabledControl,
+    onChange: disabledControl,
+    onBlur: disabledControl,
+    onClick: disabledControl,
+    onRunQuery: disabledControl,
+    onRichValueChange: disabledControl,
+    onWillApplySuggestion: disabledControl,
+    portalOrigin: disabledControl,
+    additionalPlugins: disabledControl,
+    cleanText: disabledControl,
+    syntax: disabledControl,
+    syntaxLoaded: disabledControl,
+    placeholder: {
+      control: 'text',
+    },
+    query: {
+      control: 'text',
+    },
+    disabled: {
+      control: 'boolean',
+    },
+  },
```

Comment:
I think using `exclude` in the `controls` makes it cleaner and also makes sure that these controls are not removed from the props table in the documentation tab. Also we can remove the `placeholder` and `disabled` `argTypes` because storybook can infer those

```suggestion
  parameters: {
    controls: {
      exclude: [
        'onTypeahead',
        'onChange',
        'onBlur',
        'onClick',
        'onRunQuery',
        'onRichValueChange',
        'onWillApplySuggestion',
        'portalOrigin',
        'additionalPlugins',
        'cleanText',
        'syntax',
        'syntaxLoaded',
      ],
    },
  },
  argTypes: {
    query: {
      control: 'text',
    },
  },
```

Reply:
Thanks for the detailed feedback, I appreciate that! 🙌 

I have pushed recommended changes to the PR.